backend/src/ml_space_lambda/data_access_objects/app_configuration.py

```diff
@@ -0,0 +1,387 @@
+        # Given a list of cluster sizes, parse them into a list of ClusterSize objects
+        cluster_object_list = []
+        for size in cluster_size_list:
+            cluster_object_list.append(ClusterSize(size["name"], size["size"], size["master-type"], size["core-type"]))
+        return cluster_object_list
+
+
+class EMRAutoScaling:
+    def __init__(
+        self,
+        min_instances: int,
+        max_instances: int,
+        scale_out: ScalingModel,
+        scale_in: ScalingModel,
+    ):
+        self.min_instances = min_instances
+        self.max_instances = max_instances
+        self.scale_out = scale_out
+        self.scale_in = scale_in
+
+    def to_dict(self):
+        return {
+            "min-instances": self.min_instances,
+            "max-instances": self.max_instances,
+            "scale-out": self.scale_out.to_dict(),
+            "scale-in": self.scale_in.to_dict(),
+        }
+
+    @staticmethod
+    def from_dict(dict_object: dict) -> EMRAutoScaling:
+        return EMRAutoScaling(
+            dict_object["min-instances"],
+            dict_object["max-instances"],
+            ScalingModel.from_dict(dict_object["scale-out"]),
+            ScalingModel.from_dict(dict_object["scale-in"]),
+        )
+
+
+class ScalingModel:
+    def __init__(self, increment: int, percentage_mem_available: int, eval_periods: int, cooldown: int):
+        self.increment = increment
+        self.percentage_mem_available = percentage_mem_available
+        self.eval_periods = eval_periods
+        self.cooldown = cooldown
+
+    def to_dict(self):
+        return {
+            "increment": self.increment,
+            "percentage-mem-available": self.percentage_mem_available,
+            "eval-periods": self.eval_periods,
+            "cooldown": self.cooldown,
+        }
+
+    @staticmethod
+    def from_dict(dict_object: dict) -> ScalingModel:
+        return ScalingModel(
+            dict_object["increment"],
+            dict_object["percentage-mem-available"],
+            dict_object["eval-periods"],
+            dict_object["cooldown"],
+        )
+
+
+class EMRApplications:
+    def __init__(self, name: str):
+        self.name = name
+
+    @staticmethod
+    def to_dict(application_list: list):
+        list_of_dicts = []
+        for application in application_list:
+            list_of_dicts.append({"Name": application.name})
+        return list_of_dicts
+
+    @staticmethod
+    def extract_applications(application_list: list):
+        # Given a list of cluster sizes, parse them into a list of ClusterSize objects
+        application_object_list = []
+        for application in application_list:
+            application_object_list.append(EMRApplications(application["Name"]))
+        return application_object_list
+
+
+class ProjectCreation:
+    def __init__(self, admin_only: bool, allowed_groups: list):
+        self.admin_only = admin_only
+        self.allowed_groups = allowed_groups
```

Comment:
Should this be `allowed_users` or should we add it in addition to what you have here? We don't actually have the concept of groups at the moment.

Reply:
It's supposed to be `allowed_groups` based on the documentation provided; I was told it's not a feature right now, but it's something we plan to have in the future so I wanted to include it for now and just leave it blank and not leverage it in the frontend until it's ready.

Reply:
Yeah groups are definitely on the roadmap just wasn't sure if we wanted to support one off non-admin users but I'm fine with leaving it as is for now.